File: clean.py
#!/usr/bin/env python3
"""Command line utility to clean text of invisible watermarks and typographic characters."""

import argparse
import sys
import os
from pathlib import Path

# Allow imports from ../src
sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), '..', 'src')))
from remover import remove_watermarks  # type: ignore


def main() -> None:
    parser = argparse.ArgumentParser(
        description="Clean text of invisible watermarks and typographic characters."
    )
    parser.add_argument(
        "--file", type=Path, help="Path to input file. Reads from stdin if omitted."
    )
    parser.add_argument(
        "--no-invisibles",
        action="store_true",
        help="Do not remove invisible characters.",
    )
    parser.add_argument(
        "--no-typical",
        action="store_true",
        help="Do not normalise typographic characters.",
    )
    args = parser.parse_args()
    if args.file:
        data = args.file.read_text(encoding="utf-8")
    else:
        data = sys.stdin.read()
    result = remove_watermarks(
        data,
        remove_invisible=not args.no_invisibles,
        replace_typical=not args.no_typical,
    )
    sys.stdout.write(result)


if __name__ == "__main__":
    main()

File: clean.js
#!/usr/bin/env node
const fs = require('fs');
const path = require('path');
const { removeWatermarks } = require('../src/remover.js');

function printUsage() {
  console.error('Usage: clean.js [--file <path>] [--no-invisibles] [--no-typical]');
  process.exit(1);
}

function main() {
  let file = null;
  let removeInvisible = true;
  let replaceTypical = true;
  const args = process.argv.slice(2);
  for (let i = 0; i < args.length; i++) {
    const arg = args[i];
    switch (arg) {
      case '--file':
        file = args[++i];
        break;
      case '--no-invisibles':
        removeInvisible = false;
        break;
      case '--no-typical':
        replaceTypical = false;
        break;
      case '--help':
      case '-h':
        printUsage();
        break;
      default:
        console.error(`Unknown argument: ${arg}`);
        printUsage();
    }
  }
  const input = file ? fs.readFileSync(path.resolve(file), 'utf8') : fs.readFileSync(0, 'utf8');
  const output = removeWatermarks(input, { removeInvisible, replaceTypical });
  process.stdout.write(output);
}

if (require.main === module) {
  main();
}